adagrams/game.py

```diff
@@ -1,11 +1,148 @@
+import random
+
+def create_letter_pool():
+
+    letter_pool = []
+
+    letter_frequencies = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1}
+
+    for letter in letter_frequencies:
+
+        for i in range(0, letter_frequencies[letter]):
+
+            letter_pool.append(letter)
+
+    return letter_pool
+
+
 def draw_letters():
-    pass
+    
+    letter_pool = create_letter_pool()
+
+    hand = []
+
+    while len(hand) < 10:
+
+        num = random.randint(0, len(letter_pool)-1)
+        hand.append(letter_pool[num])
+        letter_pool.pop(num)
+
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    try:
+        anagram_attempt = word.upper()
+    except AttributeError: #Case where word is not a string
+        return False
+    
+    letters = set(anagram_attempt)
+
+    for letter in letters:
+        if letter_bank.count(letter) < anagram_attempt.count(letter) or letter not in letter_bank:
+            return False
+    
+    return True
 
 def score_word(word):
-    pass
+    #does putting score chart in FN vs as global variable have implications for performance/memory?
```

Comment:
Global vars can be minutely slower since when python looks for a variable it will check the local space before the global space. When considering global vs local, the first consideration should really be "Does anything else need to access this?" If the answer is no, we should make them local to ensure no code can access them that does not need them, ensuring nothing can alter the data unexpectedly. Many of our recommended coding practices distill down to "How can I reduce the surface area on my code where bugs could appear?"